docs/updates/5-1-2017-flow.md

```diff
@@ -0,0 +1,156 @@
+where properties could be null or racey.
+
+#### Utilities
+
+* **59%** coverage
+* **60** files
+* `yarn flow-utils` - creates a flow report for utils
+
+We started typing our utility functions. Starting with utilities helped us
+document our primary business logic and made them easier to refactor.
+
+In this example, we added types to a parser utility that checks to see if an
+expression is in a particular scope.
+
+```js
+export function isExpressionInScope(expression: string, scope?: Scope) {
+  if (!scope) {
+    return false;
+  }
+
+  const variables = getVariablesInScope(scope);
+  const firstPart = expression.split(/\./)[0];
+  return variables.includes(firstPart);
+}
+```
+
+We are not currently typing our test files, but I think it would be helpful if
+we did. In this example, we are searching for all of the functions in a given
+source, but because we did not provide a type for `getSourceText`, flow can
+not use `getSymbols`'s types to tell us if we're writing a reasonable test.
+
+```js
+it("finds functions", () => {
+  const fncs = getSymbols(getSourceText("func")).functions;
+  const names = fncs.map(f => f.value);
+  expect(names).to.eql(["square", "child", "anonymous"]);
+});
+```
+
+Overall, typing our utilities was a great first step. And the more that we type
+the components, actions, and reducers that call the utilities, the more value
+we will get out of them. Already, the types are forcing an internal consistency
+that helps us achieve a better API for our utils.
+
+#### Actions / Reducers
+
+Our Actions and Reducers are generally very well covered.
+
+* **78%** coverage
+* **19** files
+* `yarn flow-redux` - creates a flow report for actions and reducers
+
+Actions like breakpoints and pause have a coverage score of over 80%.
+We initially, typed our action dispatcher params. We've recently added
+types for `ThunkArgs` and `State`. Typing `State`, allows flow to type check
+our selectors and utils.
+
+
+```js
+export function newSource(source: Source) {
+  return ({ dispatch, getState }: ThunkArgs) => {
+    if (prefs.clientSourceMapsEnabled) {
+      dispatch(loadSourceMap(source));
+    }
+
+    dispatch({ type: constants.ADD_SOURCE, source });
+
+    checkSelectedSource(getState(), dispatch, source);
+    checkPendingBreakpoints(getState(), dispatch, source);
+  };
+}
+```
+
+Our reducers are 80% covered and some reducers like breakpoints and pause are 80+
+percent covered. Breakpoints is actually 97% covered. Most of the coverage comes
+from our State type and Action union types.
+
+The `State` type describes the state of the the reducer store. Typing the State
+and action types lets us fully describe our reducers and get a high flow coverage
+score.
+
+```js
+export type BreakpointsState = {
+  breakpoints: I.Map<string, Breakpoint>,
+  pendingBreakpoints: any[],
+  breakpointsDisabled: false
+};
```

Comment:
There are still a lot of `any`s, especially for callback functions and items stored in Immutable structures. Might be good to either review if they are trivially removable or, if there are real impediments, make a list of what they are.